folium/folium.py

```diff
@@ -508,24 +522,28 @@ def choropleth(self, geo_data, data=None, columns=None, key_on=None,
         ...              columns=['Data 1', 'Data 2'],
         ...              key_on='feature.properties.myvalue',
         ...              fill_color='PuBu',
-        ...              threshold_scale=[0, 20, 30, 40, 50, 60])
+        ...              bins=[0, 20, 30, 40, 50, 60])
         >>> m.choropleth(geo_data='countries.json',
         ...              topojson='objects.countries')
         >>> m.choropleth(geo_data='geo.json', data=df,
         ...              columns=['Data 1', 'Data 2'],
         ...              key_on='feature.properties.myvalue',
         ...              fill_color='PuBu',
-        ...              threshold_scale=[0, 20, 30, 40, 50, 60],
+        ...              bins=[0, 20, 30, 40, 50, 60],
         ...              highlight=True)
 
         """
-        if threshold_scale is not None and len(threshold_scale) > 6:
-            raise ValueError('The length of threshold_scale is {}, but it may '
-                             'not be longer than 6.'.format(len(threshold_scale)))  # noqa
         if data is not None and not color_brewer(fill_color):
             raise ValueError('Please pass a valid color brewer code to '
                              'fill_local. See docstring for valid codes.')
 
+        if 'threshold_scale' in kwargs:
+            if kwargs['threshold_scale'] is not None:
+                bins = kwargs['threshold_scale']
+            raise DeprecationWarning('''
+                choropleth `threshold_scale` parameter is now depreciated
+                in favor of the `bins` parameter.''')
+
```

Comment:
I am not sure that:
```python
            raise DeprecationWarning('''
                choropleth `threshold_scale` parameter is now depreciated
                in favor of the `bins` parameter.''')
```

Is the ideal way to do it as it will stop the script as an error.

I have looked online and tried:
```python
warnings.warn('''
                choropleth `threshold_scale` parameter is now depreciated
                in favor of the `bins` parameter.''', DeprecationWarning)
```

But it doesn't seem to be shown in the Notebook I am using (it works well in the unit test on the other hand).

I am sure you have a suggestion :)

Reply:
You're right that the warning shouldn't be raised, instead `warnings.warn(msg, category)` is the way to go. Good find.

Deprecation warnings are ignored by default. They only appear in systems that enable them. We could use a `FutureWarning`, but I think I'm more inclined to use a `DeprecationWarning` and then once we later really remove the feature raise an error.
https://docs.python.org/3/library/warnings.html#warning-categories

Reply:
I have updated the code, we should be good